integration_tests/run_pyspark_from_build.sh

```diff
@@ -117,10 +117,24 @@ else
     NUM_LOCAL_EXECS=${NUM_LOCAL_EXECS:-0}
     MB_PER_EXEC=${MB_PER_EXEC:-1024}
     CORES_PER_EXEC=${CORES_PER_EXEC:-1}
-
-    if ((NUM_LOCAL_EXECS > 0)); then
-      export PYSP_TEST_spark_master="local-cluster[$NUM_LOCAL_EXECS,$CORES_PER_EXEC,$MB_PER_EXEC]"
+    # Spark 3.1.1 includes https://github.com/apache/spark/pull/31540
+    # which helps with spurious task failures as observed in our tests. If you are running
+    # Spark versions before 3.1.1, this sets the spark.max.taskFailures to 4 to allow for
+    # more lineant configuration, else it will set them to 1 as spurious task failures are not expected
+    # for Spark 3.1.1+
+    set +e
+    VERSION_PROPERTY=`unzip -q -c $SPARK_HOME/jars/spark-core_*.jar spark-version-info.properties | grep ^version=`
+    if [[ $? -eq "0" ]]; then
+      VERSION_STRING=`echo "$VERSION_PROPERTY" | cut -d'=' -f2`
```

Comment:
This could be simplified a bit by just letting the pipeline try to extract the value and then check for an empty result, e.g.:
```
VERSION_STRING=`unzip -q -c $SPARK_HOME/jars/spark-core_*.jar spark-version-info.properties | grep ^version= | cut -d'=' -f2`
[[ -z $VERSION_STRING ]] && { echo "Unable to detect the Spark version at $SPARK_HOME"; exit 1; }
echo "Detected Spark version $VERSION_STRING"
```